### PR TITLE
Refactored out the container builder to be more flexible

### PR DESCRIPTION
**Pull request by exts**
I made it so it's easier to use other containers within the application, so i refactored out the hard-coded php-di stuff into a default container class and added a new container builder interface. No breaking changes and backwards compatibility should work out of the box.

No tests failed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 vendor
+.idea/
```
Comment (Brammm) [inline .gitignore]: I personally have this in my global composer ignore. I'm not a big fan of putting IDE specific things in gitignore.

File: composer.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
         }
     ],
     "require": {
+        "php": "^7.0",
```
Comment (Brammm) [inline composer.json]: Should be `7.1` as void return types are used.

File: src/Container/BuilderInterface.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @package Starch\Container
+ */
+interface BuilderInterface
+{
+    /**
+     * @return ContainerInterface
+     */
+    public function build() : ContainerInterface;
+}
```
Comment (Brammm) [inline src/Container/BuilderInterface.php]: Missing a new line

File: src/Container/DefaultContainerBuilder.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $this->builder->addDefinitions([
+            EmitterInterface::class => object(SapiEmitter::class),
+            InvokerInterface::class => get(Container::class),
+            StackInterface::class => object(Stack::class),
+        ]);
+
+        return $this->builder->build();
+    }
+}
```
Comment (Brammm) [inline src/Container/DefaultContainerBuilder.php]: Missing a new line

**Comment by exts** [comment]
let me change them in a bit

**Comment by exts** [comment]
If you decide not to accept this approach, could you add a open source license like the MIT so I can reuse your code.

**Comment by Brammm** [comment]
@exts I've created this in response: https://github.com/starchphp/starch/pull/3

Let me know if that approach works for you.

**Comment by Brammm** [comment]
Closing in favor of https://github.com/starchphp/starch/pull/3

**Comment by Brammm** [comment]
@exts I hope I didn't come across as too ungrateful. This is the first time I've put my code out there and you were to first to interact with it. I really want to thank you for giving my project a look and chance.

**Pull request closed without merging**